sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileScanRDD.scala

```diff
@@ -230,6 +299,10 @@ class FileScanRDD(
 
       /** Advances to the next file. Returns true if a new non-empty iterator is available. */
       private def nextIterator(): Boolean = {
+        // This function is called at the end of each file and before the first file. Therefore,
+        // we update the Variant Metrics at the end of each JSON file and reset the
+        // variant metrics objects to process the next file.
+        updateVariantBuilderMetrics()
```

Comment:
Can we make this API more generic, like adding an update api on `FileScanMetrics`?

```
fileScanMetrics.update()
```

Ideally, `FileScanRDD` doesn't need to know anything about variants, and all the updating and variant specific things happens in `FileScanMetrics`.